beets/dbcore/types.py

```diff
@@ -117,6 +117,14 @@ def to_sql(self, model_value):
         """
         return model_value
 
+    def __eq__(self, other):
+        """Allow to compare type instances by their full namespaced name
+        """
+        def get_ns(o):
+            return "{m}.{n}".format(m=o.__class__.__module__,
+                                    n=o.__class__.__name__)
+        return get_ns(self) == get_ns(other)
```

Comment:
This implementation of equity will make any two objects of the same type compare equal. That's not quite what we want—while we would like any object constructed with `Integer()` to be equal to any other, we would not want `PaddedInt(4)` to be equal to `PaddedInt(2)`. So unfortunately, I think there is no way around providing per-class equality functions.

For parameter-free classes like `Integer`, the implementation could do this to avoid stringifying things:

```py
def __eq__(self, other):
    if not isinstance(other, type(self)):
        return NotImplemented
    return True
```

Reply:
I see your point. In fact, I also thought about the classes which are initialised with a length parameter after I created this PR. 

So, what you are saying is that the comparison needs to be created for each class and in the case of classes like Float or PaddedInt (with length parameter) also the length will need to be object of the comparison.

Q: in your example, why are you returning an error instead of simply returning `False`?

Reply:
`NotImplemented` is a special value that says that the comparison is not valid—for example, you're not supposed to do `'a' == 5` in Python, and it should be an error rather than simply saying "no." This blog post has some useful background:
https://eev.ee/blog/2012/03/24/python-faq-equality/

In hindsight, we should probably only do `NotImplemented` for things that are not `Type`s. We should use `False` for two things that are both `Type` instances but not of the same subclass.

Reply:
Ok. In the light of this I think that we will need to test each `Type` separately.